include/seqan3/core/detail/template_inspection.hpp

```diff
@@ -235,6 +236,36 @@ struct valid_template_spec_or<fallback_t, templ_t, spec_t...>
 template <typename fallback_t, template <typename ...> typename templ_t, typename ...spec_t>
 using valid_template_spec_or_t = typename valid_template_spec_or<fallback_t, templ_t, spec_t...>::type;
 
+// ----------------------------------------------------------------------------
+// template_specialisation_of
+// ----------------------------------------------------------------------------
+
+/*!\addtogroup concept
+ * \{
+ */
+
+/*!\interface seqan3::template_specialisation_of <>
+ * \brief Provides concept `seqan3::template_specialisation_of<mytype, [...]>` for checking the type specialisation of
+ *        some type with a given template, for example a specialized `type_list<float>` with the `type_list` template.
+ *
+ * \ingroup type_traits
+ *
+ * \tparam mytype           The source type.
```

Comment:
```suggestion
 * \tparam mytype           The guery type.
```

? 
Source type somehow threw me off. Maybe query isn't the best alternative thought. `The type to check` ?
Source type sounds like I'm actually doing something with it or taking something from it to put somewhere else..